src/resolvers/tag.ts

```diff
@@ -0,0 +1,70 @@
+import { Resolvers } from "../types";
+import { MyContext } from "../context";
+import { Section } from "../models/Section";
+import { Tag } from "../models/Tag";
+import { hasPermission } from "../services/sectionService";
+import { NotFoundError, ForbiddenError } from "../utils/graphQLErrors";
+
+
+export const resolvers: Resolvers = {
+    Query: {
+        tags: async (_, __, context: MyContext): Promise<Tag[]> => {
+            return await Tag.getAllTags('tags resolver', context);
+        },
+        tagsBySectionId: async (_, { sectionId }, context: MyContext): Promise<Tag[]> => {
+            // Find section with matching sectionId
+            const section = await Section.getSectionWithTagsBySectionId('section resolver', context, sectionId);
+            if (!section) {
+                throw NotFoundError('Section not found')
+            }
+
+            if (await hasPermission(context, section.templateId)) {
+                return await Tag.getTagsBySectionId('tagsBySectionId resolver', context, sectionId);
+            }
+
+            throw ForbiddenError();
+        },
+    },
+    Mutation: {
+        addTag: async (_, { name, description }, context: MyContext): Promise<Tag> => {
```

Comment:
I wasn't sure who has permissions to add and update tags, so I didn't add any checks for roles or permissions.

Reply:
I think this will only be SuperAdmin users, but it will never be allowed for Researchers so I think we can add a check for SuperAdmin for now.